openquake/hazardlib/gsim/kanno_2006.py

```diff
@@ -0,0 +1,517 @@
+    #: of two orthogonal horizontal components in the time domain" (p. 880)
+    DEFINED_FOR_INTENSITY_MEASURE_COMPONENT = const.IMC.PEAK_SRSS_HORIZONTAL
+
+    #: Although interevent and intraevent residuals are separately discussed
+    #: in the context of focal depth and site conditions, only the total
+    #: standard deviation is tabulated.
+    DEFINED_FOR_STANDARD_DEVIATION_TYPES = set([const.StdDev.TOTAL])
+
+    #: "Coefficients p and q were derived by regression analysis on the
+    #: residuals averaged at intervals of every 100 m/sec in AVS30." (p. 884)
+    REQUIRES_SITES_PARAMETERS = set(('vs30',))
+
+    #: Sole required rupture parameter is magnitude; faulting style is not
+    #: addressed.
+    REQUIRES_RUPTURE_PARAMETERS = set(('mag',))
+
+    #: "The source distance is the closest distance from a fault plane to the
+    #: observation site and is the hypocentral distance in the case of
+    #: earthquakes for which the fault model is not available." (p. 880)
+    REQUIRES_DISTANCES = set(('rrup',))
+
+    def get_mean_and_stddevs(self, sites, rup, dists, im_type, stddev_types):
+        # pylint: disable=too-many-arguments
+        """
+        See :meth:`superclass method
+        <.base.GroundShakingIntensityModel.get_mean_and_stddevs>`
+        for specification of input and result values.
+
+        Implements the following equations:
+
+        Equation (5) on p. 881 predicts ground motion for shallow events
+        (depth <= 30 km):
+
+        ``log(pre) = a*M + b*X - log(X + d*10^(e*M)) + c + epsilon``
+
+        "where pre is the predicted PGA (cm/sec^2), PGV (cm/sec), or 5%
+        damped response spectral acceleration (cm/sec^2)" (p. 883) and a,
+        b, c and d are tabulated regression coefficients. Note that
+        subscripts on the regression coeffients have been dropped -
+        subscript `1` denoted "shallow" while subscript `2` denoted "deep"
+        - so that the "deep" model of equation (6) can be implemented trivally
+        by changing coefficients and setting d = 0.
+
+        Equation (8) on p. 883 gives the model used for site amplitfication:
+
+        ``G = p*log(VS30) + q``
+
+        Where p and q are tabulated regression coefficients.
+
+        Equation (9) on p. 884 for the ground motion at a given site:
+
+        ``log(pre_G) = log(pre) + G``
+
+        No adjustment of epsilon is made as a function of VS30.
+
+        Note finally that "log represents log_10 in the present study"
+        (p. 880).
+
+        """
+
+        assert im_type.__class__ in self.DEFINED_FOR_INTENSITY_MEASURE_TYPES
+
+        # obtain coefficients for required intensity measure type (IMT)
+        coeffs = self.COEFFS_BASE[im_type].copy()
+        coeffs.update(self.COEFFS_SITE[im_type])
+
+        # obtain IMT-independent coefficients
+        coeffs.update(self.CONSTS)
+
+        # raise warning outside applicable range
+        is_valid = ~((dists.rrup < coeffs['R_valid']) &
```

Comment:
In general we don't raise warning for GMPEs used outside of their ``applicable'' range. This is something we have discussed a lot in GEM over the years and there are arguments for and against it (personally I'm against it). If we were to do it then we will introduce a more systematic approach that would apply across all GMPEs rather than one.